File: Lista 1/33.c
#include <stdio.h>

int main()
{
    int id;
    int quantity = 0;
    double grade;
    double total = 0.0;

    while (1)
    {
        printf("Entry student Id or 0 to exit:\n");
        scanf("%d", &id);
        if (id == 0)
        {
            break;
        }

        printf("Entry the grade:\n");
        scanf("%lf", &grade);

        total += grade;
        quantity++;

    }

    // Debug
    printf("id: %d\n", id);
    printf("total: %lf\n", total);
    printf("quantity: %d\n", quantity);

    if (quantity > 0)
    {
        printf("The grade is: %.2lf\n", total / quantity);
    }
    else
    {
        printf("No students.\n");
    }

    return 0;
}

File: Lista 1/33.py
""""
    Escreva um programa que calcule e exiba a média da nota dos alunos de uma turma em uma prova. O número de 
    alunos é desconhecido. Os dados de um aluno são: número de matrícula e a sua nota na prova em questão.
"""

student_id_data = []
student_grade_data = []
result = 0;

while(True):
    student_id = input("Entry student Id or /0 to exit: ")
    if student_id == "/0":
        break;
    student_id_data.append(student_id)
    student_grade = int(input("Entry studenty grade: "))
    student_grade_data.append(student_grade)

for item in student_grade_data:
    result += student_grade

print(f"The media is: {result / len(student_grade_data)}")